File: Algorithms/Greedy/Priyanka and Toys.php
<?php

/*
 * Complete the 'toys' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY w as parameter.
 */

function toys($w) {
    # Write your code here
    sort($w);
    $count = 0;
    $i = 0;
    while ($i < count($w)) {
        $j = $i;
        while ($j < count($w) && $w[$j] - $w[$i] <= 4) {
            $j++;
        }
        $count++;
        $i = $j;
    }
    return $count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));
$w_temp = rtrim(fgets(STDIN));
$w = array_map('intval', preg_split('/ /', $w_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = toys($w);

fwrite($fptr, $result . "\n");
fclose($fptr);

File: Algorithms/Greedy/Priyanka and Toys.py
import os

#
# Complete the 'toys' function below.
#
# The function is expected to return an INTEGER.
# The function accepts INTEGER_ARRAY w as parameter.
#

def toys(w):
    # Write your code here
    w.sort()
    i = 0
    count = 0
    while i < len(w):
        j = i
        while j < len(w) and w[j] - w[i] <= 4:
            j += 1
        count += 1
        i = j
    return count


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())
    w = list(map(int, input().rstrip().split()))

    result = toys(w)

    fptr.write(str(result) + '\n')
    fptr.close()
